File: app/src/main/java/com/elli0tt/rpg_life/presentation/character/CharacteristicsViewModel.java
package com.elli0tt.rpg_life.presentation.character;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.elli0tt.rpg_life.data.repository.CharacteristicsRepositoryImpl;
import com.elli0tt.rpg_life.domain.modal.Characteristic;

import java.util.ArrayList;
import java.util.List;

public class CharacteristicsViewModel extends AndroidViewModel {
    private CharacteristicsRepositoryImpl characteristicsRepository;
    private LiveData<List<Characteristic>> allCharacteristics;

    public CharacteristicsViewModel(@NonNull Application application) {
        super(application);
        characteristicsRepository = new CharacteristicsRepositoryImpl(application);
        allCharacteristics = characteristicsRepository.getAllCharacteristics();
    }

    public LiveData<List<Characteristic>> getAllCharacteristics() {
        return allCharacteristics;
    }

    public void insert(Characteristic characteristic){
        characteristicsRepository.insert(characteristic);
    }

    private List<Characteristic> generateSampleCharacteristicsList() {
        List<Characteristic> resultList = new ArrayList<>();
        resultList.add(new Characteristic("Strength"));
        resultList.add(new Characteristic("Intelligence"));
        resultList.add(new Characteristic("Agility"));
        resultList.add(new Characteristic("Endurance"));
        resultList.add(new Characteristic("Willpower"));
        resultList.add(new Characteristic("Procrastination"));
        resultList.add(new Characteristic("Self-confidence"));
        resultList.add(new Characteristic("Communication"));
        return resultList;
    }

    public void populateWithSamples(){
        characteristicsRepository.insert(generateSampleCharacteristicsList());
    }
}

File: app/src/main/java/com/elli0tt/rpg_life/presentation/character/CharacteristicsViewModel.kt
package com.elli0tt.rpg_life.presentation.character

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import com.elli0tt.rpg_life.data.repository.CharacteristicsRepositoryImpl
import com.elli0tt.rpg_life.domain.model.Characteristic
import java.util.*

class CharacteristicsViewModel(application: Application) : AndroidViewModel(application) {
    private val characteristicsRepository: CharacteristicsRepositoryImpl = CharacteristicsRepositoryImpl(application)
    private val allCharacteristics: LiveData<List<Characteristic>>

    init {
        allCharacteristics = characteristicsRepository.allCharacteristics
    }

    fun insert(characteristic: Characteristic?) {
        characteristicsRepository.insertCharacteristics(characteristic)
    }

    private fun generateSampleCharacteristicsList(): List<Characteristic> {
        val resultList: MutableList<Characteristic> = ArrayList()
        resultList.add(Characteristic("Strength"))
        resultList.add(Characteristic("Intelligence"))
        resultList.add(Characteristic("Agility"))
        resultList.add(Characteristic("Endurance"))
        resultList.add(Characteristic("Willpower"))
        resultList.add(Characteristic("Procrastination"))
        resultList.add(Characteristic("Self-confidence"))
        resultList.add(Characteristic("Communication"))
        return resultList
    }

    fun populateWithSamples() {
        characteristicsRepository.insertCharacteristics(generateSampleCharacteristicsList())
    }
}